src/lib/viewers/doc/PresentationPreloader.js

```diff
@@ -24,6 +29,28 @@ class PresentationPreloader extends DocPreloader {
     }
 
     /**
+     * Get dimensions for the preload image. For presentations, this is always scaled to fit inside the viewport since
+     * pdf.js auto-scales presentations to fit the window with no maximum zoom.
+     *
+     * @param {number} pdfWidth - Width of PDF in pixels
+     * @param {number} pdfHeight - Height of PDF in pixels
+     * @return {Object} Width and height in pixels
+     */
+    getScaledDimensions(pdfWidth, pdfHeight) {
```

Comment:
If this is identical to DocPreloader's, can you move it to `util.js`, with an additional parameter for `wrapperEl` OR `clientWidth` and `clientHeight`?

Reply:
I guess there's a slight difference. Is there anything you can abstract to remove duplicated code?

Reply:
Hmm yeah, can probably make max scale an optional property and keep it in docpreloader. Good point, thanks.